### DISCUSSION TITLE
await data of useAsyncData

### DISCUSSION THREAD

**Discussion by feelixe** [nuxt/nuxt#16551] [Questions] [answered]
In this example
```
const { data, pending, refresh } = await useAsyncData(() => $fetch("/api/hello/"))
```
It would be nice to have the ability to await the promise returned by the handler to be resolved. So that you can render the template after the data is set and you don't have to check `pending` and `error` and just use `Suspense` in the parent component.

Is there any way to achieve this?

**Comment by TheAlexLichter** [comment, marked as answer]
This should already be the case when not using the lazy version! See [this stackblitz](https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions8463?file=app.vue)

When fetching on the client only, be aware of [this limitation](https://v3.nuxtjs.org/api/composables/use-async-data/#return-values).

> If you have not fetched data on the server (for example, with `server: false`), then the data *will not* be fetched until hydration completes. This means even if you await `useAsyncData` on the client side, `data` will remain null within `<script setup>`.

**Comment by feelixe** [comment]
I see, there seems to be some problem with client-rendered components. Check this [stackblitz](https://stackblitz.com/edit/github-u3ej3d?file=package.json).
In the stackblitz you get 500 error because the data has not been set. However, if you try rendering just the `ServerComponent` it works as expected.

I've tried adding `{server: false}` to the `useAsyncData` options on the `.client` component, but that does not help.

**Reply by TheAlexLichter** [reply]
See the warning on https://v3.nuxtjs.org/api/composables/use-async-data/#return-values regarding fetching only on the client ☺️

**Reply by feelixe** [reply]
Ahhhh, I see, thank you very much💛

**Reply by TheAlexLichter** [reply]
No problem ☺️
